### PR TITLE
optimize: update the LICENSE and NOTICE files and standardize dependency versions

### PR DESCRIPTION
**Pull request by funky-eyes**
- [ ] I have registered the PR [changes](../changes).

### CODE REVIEW

**Review by xingfudeshi — APPROVED** [review]
LGTM

File: build/pom.xml [hunk 1]
```diff
@@ -81,8 +81,8 @@
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
         <!-- The version of spring-boot for 'spring-boot-dependencies' and 'spring-boot-maven-plugin' -->
-        <spring-boot.version>2.5.15</spring-boot.version>
-        <spring-framework.version>5.3.27</spring-framework.version>
+        <spring-boot.version>2.7.18</spring-boot.version>
```
Comment (slievrly) [inline build/pom.xml:84]: The Spring dependencies for the client and server have been version-separated. The scope of the Spring version on the client side is set to "Provided," and it is maintained by the business side, allowing for the use of lower versions to ensure compatibility with more development frameworks. The server, seata-server, is independently deployed and has been upgraded to a higher version to address security vulnerabilities.

Comment (funky-eyes) [reply build/pom.xml:84]: > The Spring dependencies for the client and server have been version-separated. The scope of the Spring version on the client side is set to "Provided," and it is maintained by the business side, allowing for the use of lower versions to ensure compatibility with more development frameworks. The server, seata-server, is independently deployed and has been upgraded to a higher version to address security vulnerabilities.

The Spring API used by the Seata client will not become incompatible due to this version upgrade.

**Review by slievrly — COMMENTED** [review]
some npmjs dependencies need to be check the version：

https://github.com/apache/incubator-seata/pull/6841/files#diff-47c43934aa38d6d5d37283eb11c67cf9901474a674a60fe61832037a1897ae0aR647

https://github.com/apache/incubator-seata/pull/6841/files#diff-47c43934aa38d6d5d37283eb11c67cf9901474a674a60fe61832037a1897ae0aL757

https://github.com/apache/incubator-seata/pull/6841/files#diff-47c43934aa38d6d5d37283eb11c67cf9901474a674a60fe61832037a1897ae0aL620

https://github.com/apache/incubator-seata/pull/6841/files#diff-47c43934aa38d6d5d37283eb11c67cf9901474a674a60fe61832037a1897ae0aL622

**Pull request merged**